index.d.ts

```diff
@@ -114,6 +114,7 @@ type FeaturedPostsProps = {
 
 type PostPreviewProps = {
```

Comment:
I think it would be better to keep component prop types inside the respective component files, and keep `index.d.ts` for more global types, like data from the CMS. That way component props are easier to manage, I think.